csharp/Migrate.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Linq;
+using Cassandra; // DataStax Cassandra C# driver
+using Microsoft.Extensions.Logging;
+
+namespace CarePet
+{
+    public class Migrate
+    {
+        private static readonly ILogger<Migrate> LOG;
+        private readonly Config _config;
+
+        static Migrate()
+        {
+            // You can wire up any logger provider here (Console, Serilog, etc.)
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddConsole();
+            });
+            LOG = loggerFactory.CreateLogger<Migrate>();
+        }
+
+        public Migrate(Config config)
+        {
+            _config = config;
+        }
+
+        public static void Main(string[] args)
+        {
+            var config = Config.Parse(new Config(), args);
+
+            var client = new Migrate(config);
+            client.CreateKeyspace();
+            client.CreateSchema();
+            client.PrintMetadata();
+        }
+
+        /// <summary>
+        /// Initiates a connection without a specific keyspace.
+        /// </summary>
+        public ISession Connect()
+        {
+            return _config.Builder().Build().Connect();
+        }
+
+        /// <summary>
+        /// Initiates a connection with the configured keyspace.
+        /// </summary>
+        public ISession Keyspace()
+        {
+            return _config.Builder(Config.Keyspace).Build().Connect(Config.Keyspace);
```

Comment:
This one does not make sense, you either provide keyspace to builder or to connect.